### PR TITLE
ref(typing): Add additional stub packages for type checking

### LINKED ISSUE #2226: Start using `types-WebOb`
### Problem Statement

I've recently contributed type stubs for WebOb on typeshed, which have been merged and are now available to the public. Some of the Integrations refer to types defined in WebOb (e.g. `PyramidIntegration`), but don't yet make use of `types-WebOb` and as such are really operating on `Any`.

### Solution Brainstorm

Install `types-WebOb` and fix any uncovered typing issues.

As a side note, I've noticed that sometimes you use `dict[str, str]` as the type annotation for `environ`, even though that is inaccurate. You should use `_typeshed.wsgi.WSGIEnvironment` which is currently an alias for `dict[str, Any]` but will ensure that your type hints for `environ` will remain consistent with the stdlib and any third party stubs managed through typeshed in the future.

The same goes for `WSGIApplication` and `StartResponse` defined within that module.

### PR DESCRIPTION
**Pull request by Daverball**
Adds `types-webob`, `types-greenlet` and `types-gevent` to linter requirements and fixes newly exposed typing issues.

Fixes #2226

### CODE REVIEW

File: sentry_sdk/profiler/continuous_profiler.py [hunk 1]
```diff
@@ -51,9 +52,10 @@
 
 
 try:
-    from gevent.monkey import get_original  # type: ignore
-    from gevent.threadpool import ThreadPool  # type: ignore
+    from gevent.monkey import get_original
+    from gevent.threadpool import ThreadPool as _ThreadPool
 
+    ThreadPool = _ThreadPool  # type: Optional[Type[_ThreadPool]]
```
Comment (sentrivana) [inline sentry_sdk/profiler/continuous_profiler.py:58]: Why is this necessary?

File: sentry_sdk/utils.py [hunk 2]
```diff
@@ -54,6 +54,8 @@
         Union,
     )
 
+    from gevent.hub import Hub
```
Comment (sentrivana) [inline sentry_sdk/utils.py:57]: Since `gevent` is an optional dependency, won't this break type checking for folks that don't have it installed? Would a conditional import help in any way?

**Review by sentrivana — COMMENTED** [review]
Thanks @Daverball, looks good to me overall. I left two comments, could you please take a look?

File: sentry_sdk/profiler/continuous_profiler.py (hunk 1 above)
Comment (Daverball) [reply sentry_sdk/profiler/continuous_profiler.py:58]: If we don't do this, then mypy will complain about redefinition of `ThreadPool` to `None` in the `except` clause. So the type and the variable needs a separate name, so I can use the type separately from the runtime variable.

File: sentry_sdk/utils.py (hunk 2 above)
Comment (Daverball) [reply sentry_sdk/utils.py:57]: There's no conditional imports for typing unfortunately. So yes, if people leave `follow_imports` at `normal` for `sentry_sdk` they will see an error if they don't have `types-gevent` installed, but the same is true for any of the other typing only depdencies that are only listed in `dev-requirements.txt`, so this would be nothing new.

Generally the expectation with type hints is that people are responsible for managing `typing-` requirements and otherwise silence the errors by setting `follow_imports` to `silent` or `skip`.

**Review by sentrivana — APPROVED** [review]
